Warn if trial_type column not found, set to None

adelavega opened the pull request:
Close #47 

If `condition_column` (by default trial_type) is not found, warn instead of error, and set `condition_column` to None,

adelavega commented:
Okay I've added two more things to this PR:
- Fix #49  by adding an option to pass in a `files_directory` folder (rather than `files`). A temporary BIDSLayout is formed to extract the entities from tsv files in this folder, and they are matched to functional runs in the original project. 
- When building a dense index, there was another bug in which frequencies under 1s would error because pandas expects values greater than 1 (perhaps event integers). Instead, specifying frequency as milliseconds and rounding to nearest ms.

adelavega commented:
Test related to reading in files is failing because change in approach. An alternative is to use a similar approach that takes a file list but puts all files into a flat temp folder. Either way, it should not be expected these files are in the original dir structure.

adelavega commented:
Finally, patsy keeps erroring on import. Tried to fix.

And, when 'amplitude_value' was set, df was not created due to indent error.

tyarkoni commented on bids/events/base.py:
```diff
@@ -250,24 +250,25 @@ def __init__(self, base_dir, default_duration=0, default_amplitude=1,
-                None (default), will use all event files found within the
-                current BIDS project. Note that if `files` is passed, existing
-                event files within the BIDS project are completely ignored.
+            file_directory (None, str): Optional path to folder containing
```
I'm not sure about this approach, because it requires all of the event files to be in the same folder, which is not at all guaranteed (e.g., I can imagine a common use case for this functionality being passing in the result of a wild card search through a directory tree). I'm wondering if maybe we should just drop support for extra-BIDS event files entirely for now. It seems like more trouble than it's worth. (Down the road, the way to address this is probably to refactor the classes in `base.py` so that they inherit from some non-BIDS-specific class that can handle arbitrary event file formats.)

tyarkoni replied:
Oh, and re: patsy, this appears to be a Windows-specific issue. Not sure what the problem is, but the travis tests are all passing as of my latest commits, without the version locking in requirements.txt, so I think it's probably fine without that, unless you're having problems with it on linux or Mac.

adelavega replied:
That's probably fine. So what do you recommend for people that need to replace the original event files (i.e. us)? Write out in place of the original files?

adelavega replied:
Also, I've had patsy error on my in Docker, when pybids was a requirement for something else.

adelavega replied:
Looks like this fixes it for me at least

adelavega replied:
I think I would prefer to keep some sort of files argument. We could also fetch the files and put them all in a temp folder, and use the same approach I'm using. Not the most efficient but I think it's a useful feature. I rather not overwrite the original dataset if I can help it.

tyarkoni replied:
Good point. Okay, I agree with your comment in #49: this probably requires that we extract entities from the incoming file. Let's go with your current approach for the moment, and I'll think about the best way to handle this more generally (likely by adding an `add_file` method to the grabbit `Layout` class).

adelavega replied:
Yeah that seems like the proper long term solution. The beauty of BIDS is all the entities are stored in the filename, right? Are there any instances where the location would matter? The the project could maintain new files as if they'd been there in the first place and they're linked to other files by matching entities.

adelavega commented:
Allright since this PR is still out here, I added two more things. Feedback plz.
 - Filenames were not being output in BIDS compliance because 1) order of entities was wrong and 2) "subject" was used instead of "sub". Fixed both
 - Column useful only internally were being output such as "time" and "event_file_id". Now dropping those by default (maybe should be an option?) as well as dropping entities when outputting one file per combination of entities.

adelavega commented:
ok!
